python_modules/dagster/dagster_tests/execution_tests/execution_plan_tests/test_execution_plan_reexecution.py

```diff
@@ -261,3 +261,48 @@ def test_job_step_key_subset_execution():
                 run_config=run_config,
                 instance=instance,
             )
+
+
+def define_pool_job() -> JobDefinition:
+    @op(pool="upstream_pool")
+    def upstream():
+        return 1
+
+    @op(pool="downstream_pool")
+    def downstream(inp):
+        return inp
+
+    @job
+    def pool_job():
+        downstream(upstream())
+
+    return pool_job
+
+
+def test_pool_reexecution():
+    with instance_for_test() as instance:
+        # initial execution
+        with execute_job(reconstructable(define_pool_job), instance=instance) as result:
+            parent_run = result.dagster_run
+            assert parent_run.run_op_concurrency
+            assert parent_run.run_op_concurrency.all_pools == {
+                "upstream_pool",
+                "downstream_pool",
+            }
+            assert parent_run.run_op_concurrency.root_key_counts == {"upstream_pool": 1}
+            assert parent_run.status == DagsterRunStatus.SUCCESS
+
+        # retry execution
+        with execute_job(
+            reconstructable(define_pool_job),
+            instance=instance,
+            reexecution_options=ReexecutionOptions(
+                parent_run_id=parent_run.run_id,
+                step_selection=["downstream"],
+            ),
+        ) as result:
```

Comment:
is it worth testing a case where its not a retry but there's still a step selection? i guess those play out pretty identically as far as this code is concerned